Use AWS SDK to retrieve ec2 instance identity doc

mcpherrinm opened the pull request:
This is an alternative design to #1360 
Rather than trying to always use the SDK, the fallback path here uses the old code.
This is thus more compatible, at the expense of being a bit branchier.

Which issue this PR fixes
fixes #1359

azdagron commented on pkg/agent/plugin/nodeattestor/aws/iid.go:
```diff
@@ -36,8 +40,9 @@ func builtin(p *IIDAttestorPlugin) catalog.Plugin {
-	IdentityDocumentURL  string `hcl:"identity_document_url"`
-	IdentitySignatureURL string `hcl:"identity_signature_url"`
+	EC2MetadataEndpoint  string `hcl:"ec2_metadata_endpoint"`
+	IdentityDocumentURL  string `hcl:"identity_document_url"`  // Deprecated
```
nit: prefix vars with `Deprecated` (but keep hcl tag as-is) as a visual indicator that new code shouldn't rely on them

azdagron commented on pkg/agent/plugin/nodeattestor/aws/iid.go:
```diff
@@ -105,12 +153,15 @@ func (p *IIDAttestorPlugin) Configure(ctx context.Context, req *spi.ConfigureReq
-		config.IdentityDocumentURL = defaultIdentityDocumentURL
-	}
+	// If we have a legacy config, ensure both have a value
+	if config.EC2MetadataEndpoint == "" && (config.IdentitySignatureURL != "" || config.IdentityDocumentURL != "") {
```
We should log here if deprecated configurables are set and the new endpoint configurable is:
1) set  (e.g. "deprecated configurable foo is ignored because...")
2) not set  (.e.g. "deprecated configurable foo will be deprecated in a future release. please use ec2_metadata_endpoint instead")

mcpherrinm commented:
Addressed feedback (warning logs, use Deprecated prefix on variable names)

I rebased on master and force-pushed the updated commit since there was some merge conflicts with my other cleanup PR in #1398

mcpherrinm commented on pkg/agent/plugin/nodeattestor/aws/iid.go:
```diff
@@ -3,27 +3,32 @@ package aws
 	"encoding/json"
+	"errors"
 	"fmt"
+	"github.com/hashicorp/go-hclog"
```
fixed.  Blame my IDE upgrading losing my save-actions auto-goimports :)

mcpherrinm commented on pkg/agent/plugin/nodeattestor/aws/iid.go:
```diff
@@ -3,27 +3,32 @@ package aws
 	"net/http"
 	"sync"
 
+	awsSdk "github.com/aws/aws-sdk-go/aws"
```
Changed to `awssdk`
